sdk/identity/azure_identity/src/credentials/client_assertion_credentials.rs

```diff
@@ -68,59 +80,92 @@ impl<C: ClientAssertion> ClientAssertionCredential<C> {
         assertion: C,
         options: Option<ClientAssertionCredentialOptions>,
     ) -> azure_core::Result<Self> {
+        validate_tenant_id(&tenant_id)?;
+        validate_not_empty(&client_id, "no client ID specified")?;
+        let options = options.unwrap_or_default().credential_options;
+        let endpoint = options
+            .authority_host()?
+            .join(&format!("/{tenant_id}/oauth2/v2.0/token"))
+            .with_context(ErrorKind::DataConversion, || {
+                format!("tenant_id '{tenant_id}' could not be URL encoded")
```

Comment:
Nit: I don't quote this elsewhere. Doesn't add much, really. It's not like it'll have spaces.